chapters/edo/main.tex

```diff
@@ -1,2 +1,1193 @@
+
+% TODO Perhaps a scatter of fitness and convexity for the parents.
+% TODO A hypothesis test even? Depends on the scatter, I guess.
+% TODO Should the cluster statistics just be printed in tables for each example?
+% TODO Conclude the subsection.
+
+\subsection{Comparison with DBSCAN}\label{subsec:dbscan}
+
+The extent of the capabilities EDO holds as a tool to better understand an
+algorithm are especially apparent when comparing an algorithm against another
+(or set of others) simultaneously. To compare a set of algorithms, one must
+utilise the freedom of choice in a fitness function for EDO.\ Consider two
+algorithms, \(A\) and \(B\), and some common metric between them, \(g\). Then
+their similarities and contrasts can be explored by considering the differences
+in this metric on the two algorithms,~i.e.\ using \(f=g_A-g_B\), \(f=g_B-g_A\)
+or \(f=\left|g_B-g_A\right|\) as the fitness function. Doing so can highlight
+pitfalls, edge cases or fundamental conditions for the method(s). Overall, this
+process affords a deeper level of learning about the method of interest beyond
+the traditional empirical approach of comparison on a particular example.
+
+The final example in this case study considers a comparison of \(k\)-means with
+another clustering algorithm of a different form: Density-Based Spatial
+Clustering of Applications with Noise (DBSCAN). The objective of the first part
+of this example is to find datasets for which \(k\)-means outperforms its
+alternative, DBSCAN.\ There is no concept of inertia in DBSCAN as it is based on
+density (as opposed to raw distance) and identifies outliers~\cite{Ester1996}.
+As such, a valid metric must be chosen. Again, the silhouette coefficient is one
+such metric.
+
+The standard silhouette function is used here over that defined
+in~\eqref{eq:discounted_silhouette} for two reasons. First, the relationship
+between correlation and DBSCAN has not yet been established in this study, and
+second, to simplify the final fitness function. An adjustment
+to~\eqref{eq:silhouette} must be made since the silhouette coefficient requires
+at least two clusters and DBSCAN need only cluster a subset of a dataset
+(referred to as the \emph{core~points}), labelling the remainder as outliers.
+Let \(S_k (X)\) and \(S_D (X)\) denote the silhouette coefficients of the
+clustering found by \(k\)-means and DBSCAN respectively, and let \(Z_D\) be the
+clustering found by DBSCAN. Then the \emph{\(k\)-means-preferable} fitness
+function is defined to be:
+\begin{equation}\label{eq:kmeans_preferable}
+    f(X) :=
+        \begin{cases}
+            S_k (X) - S_D (X), &\quad \text{%
+                \begin{tabular}{l}%
+                    if \(\left|Z_D\right| > 1\)
+                \end{tabular}
+            }\\
+            - \infty &\quad \ \ \text{otherwise}
+        \end{cases}
+\end{equation}
+% TODO include a statement of DBSCAN? It's quite long 
```

Comment:
I think it's ok to just point at reference text...

Reply:
Fixed ef035bb